File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, Integer> cloth = new HashMap<>();
        
        for(String[] c: clothes){
            cloth.put(c[1], cloth.getOrDefault(c[1], 0) + 1);
        }
        for(String k: cloth.keySet()){
            answer*=(cloth.get(k)+1);
        }
        return answer-1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    var answer = 1;
    const map = new Map();
    
    for(c of clothes){
        if(map.has(c[1])){
            map.set(c[1], map.get(c[1])+1)
        }else{
            map.set(c[1], 1)
        }
    }
    
    for(m of map.values()){
        answer*=(m+1)
    }
    
    return answer-1;
}